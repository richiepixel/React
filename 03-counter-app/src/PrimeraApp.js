import React from 'react';
import PropTypes from 'prop-types'
const PrimeraApp = ({ saludo, subtitulo }) => {

  return (
    <>
      <h1>{saludo}</h1>
      {/* <pre>{JSON.stringify(saludo,null,7)}</pre> */}
      <p>{subtitulo}</p>
    </>
  );
}

PrimeraApp.propTypes = {
  saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps ={
  subtitulo: 'Soy un subt√≠tulo'
}

export default PrimeraApp;