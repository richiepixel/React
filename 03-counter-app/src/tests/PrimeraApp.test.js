import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import PrimeraApp from '../PrimeraApp';

describe('Pruebas en <PrimeraApp />', () => {
  // test('debe de mostrar el mensaje Hola Richie!', () => {
  //   const saludo = 'Hola Richie!';

  //   const { getByText } = render(<PrimeraApp saludo={saludo} />);
  //   expect(getByText(saludo)).toBeInTheDocument();
  // });

  test('debe de mostrar <PrimeraApp /> correctamente', () => {
    const saludo = 'Hola Richie!';
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar el saludo enviado por props', () => {
    const saludo = 'Hola Richie';
    const subtitulo = 'Hola subtitulo';
    const wrapper = shallow(
      <PrimeraApp saludo={saludo} subtitulo={subtitulo} />
    );

    const textoParrafo = wrapper.find('p').text();
    console.log(textoParrafo);
    expect(textoParrafo).toBe(subtitulo);
  });
});
