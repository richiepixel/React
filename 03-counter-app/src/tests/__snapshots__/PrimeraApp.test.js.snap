// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola Richie!
  </h1>
  <p>
    Soy un subt√≠tulo
  </p>
</Fragment>
`;
